components/shared_server_code/src/shared/model/measurement.py

```diff
@@ -261,8 +262,12 @@ def update_measurement(self) -> None:
             self[scale].update_value_and_status()
 
     def sources_ok(self) -> bool:
-        """Return whether there are sources and none have parse or connection errors."""
+        """Return whether there are sources and none have configuration, parse, or connection errors."""
         sources = self.sources()
+        for source in sources:
```

Comment:
Does it make sense to short-circuit parse and connection errors before looping over the sources?

Reply:
I don't think it is worth it, metrics typically have only one or a handful of sources.